### DISCUSSION TITLE
A different way of organising tailwind classes

### DISCUSSION THREAD

**Discussion by ikusteu** [librocco/librocco#121] [Ideas]
I've opened up a PR adding `prettier-config-tailwindcss`. I think it's extremely convenient for being opinionated in the way the classes are sorted, giving us less to think about. Additionally sorting classes on save instead of sorting them manually provides for faster development, with the same amount of tidyness.

This benefit doesn't come without caveats though, the biggest one being that the classes need to either be inlined (written in html on `class` property) or within `<style>` block in order for the plugin to pick them up. This would require us to change the way we write the classes within a component to reap the full benefit.

Our current way of grouping classes (in variables) is convenient and, IMO, extremely tidy way of doing so, when styling react components. However, since svelte has the `<style>` tag option, everything we did thus far with the variables can be written as custom classes in `<style>` block:

**old way**
```svelte
<script>
  export let color: "primary" | "secondary"

  const baseClasses = "relative max-w-[320px]"
  const colorPrimary = "bg-blue-300 hover:bg-blue-400"
  const colorSecondary = "bg-red-200 hover:bg-red-300"
</script>

<button class={[baseClasses, color === "primary" ? colorPrimary : colorSecondary].join(" ")} />
```

**proposed new way**
```svelte
<script>
  export let color: "primary" | "secondary"
</script>

<button class="button-base {color === "primary" ? "color-primary" : "color-secondary"}" />

<style>
.button-base {
  @apply relative max-w-[320px]
}
.color-primary {
  @apply bg-blue-300 hover:bg-blue-400
}
.color-secondary {
  @apply bg-red-200 hover:bg-red-300
}
</style>
```

As you can see, there's not much difference in the code written, both solutions are equally tidy, but we're not using JS for something as simple as storing css classes (as we already have css for that), with the added benefit of tailwind classes after `@apply` directive get sorted as if they've been written directly in button's `class` property.

What are your thoughts on this?

If you're ok with the proposed way of doing things, I'm not saying we have to rewrite the existing components (even though, our codebase being rather small still, that would take 1-2hrs tops), but to apply the new approach to new stuff we write and, maybe, incrementally rewrite the existing components.

**Comment by delasous** [comment]
Really like this and totally agree with it! I think I will also apply this approach in Edible.
